openadapt/strategies/mixins/layout.py

```diff
@@ -0,0 +1,36 @@
+"""
+Implements a ReplayStrategy mixin for extracting layout and important
+information from text documents and GUI images.
+
+Usage:
+
+    class MyReplayStrategy(LayoutExtractionReplayStrategyMixin):
+        ...
+"""
+from PIL import Image
+from typing import List
+from transformers import pipeline
+import numpy as np
+
+from openadapt.models import Recording, Screenshot
+from openadapt.strategies.base import BaseReplayStrategy
+
+
+class LayoutExtractionReplayStrategyMixin(BaseReplayStrategy):
+
+    def __init__(self, recording: Recording, image_file_paths: List[str]):
+        super.__init__(recording)
+        self.image_list = [
+            Image.open(img_file_path).convert("RGB")
+            for img_file_path in image_file_paths
+        ]
+
+    def document_query(self, image: np.ndarray, question: str) -> str:
+        query_pipeline = pipeline(
+            "document-question-answering", 
+            model="impira/layoutlm-document-qa"
+        )
+        output = query_pipeline(image, question)
+        if output != []:
```

Comment:
How about `if not output`?

Reply:
Good point, I was using ```if output is not None``` initially but I remembered from way back in first year that there's a difference between ```is not ``` and ```==/!=``` but I didn't think of using the boolean evaluation. I'll make this change 👍

Reply:
actually I tried this change out just now, and it still throws the ```IndexError: list index out of range``` whenever it returns an empty list (i.e. when the model has no answer to provide due to an unsupported document, in case of the calculator GUI screenshot for example) and so keeping it as ```if output != []``` is better since we can then raise a ```TypeError: Unsupported document...``` error